### PR TITLE
Modifying Dockerfile to pull in multicloud support

### PR DESCRIPTION
**Pull request by PratMis**
Adding commands to include Azure and GCP in the Dockerfile

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,9 +2,21 @@ FROM python:2.7
...
+WORKDIR /src/tools/c7n_azure
+RUN pip install -r requirements.txt -e .
+RUN python setup.py develop
+
+# Install Custodian GCP
+WORKDIR /src/tools/c7n_gcp
+RUN pip install -r requirements.txt -e .
+RUN python setup.py develop
+
+RUN pip install future
```
Comment (kapilt) [inline Dockerfile]: fwiw, the c7n-org tools has a good canonical dockerfile, the future install here looks odd.

**Comment by kapilt** [comment]
sigh.. looks like something is up from the upload to codecov, s3 incomplete read error

File: Dockerfile (hunk 1 above)
Comment (PratMis) [reply Dockerfile]: Thanks! I looked into the c7n-org dockerfile. Looks like c7n_azure needs this module which isn't present in any of the base docker images i.e.(pypy2 or python2.7). This is the error that pops up:
```Traceback (most recent call last):
  File "/usr/local/bin/custodian", line 11, in <module>
    load_entry_point('c7n', 'console_scripts', 'custodian')()
  File "/src/c7n/cli.py", line 368, in main
    command(config)
  File "/src/c7n/commands.py", line 349, in schema_cmd
    load_resources()
  File "/src/c7n/resources/__init__.py", line 99, in load_resources
    resources.load_plugins()
  File "/src/c7n/registry.py", line 117, in load_plugins
    f = ep.load()
  File "/usr/local/site-packages/pkg_resources/__init__.py", line 2346, in load
    return self.resolve()
  File "/usr/local/site-packages/pkg_resources/__init__.py", line 2352, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/src/tools/c7n_azure/c7n_azure/entry.py", line 15, in <module>
    from c7n_azure.provider import resources
  File "/src/tools/c7n_azure/c7n_azure/provider.py", line 20, in <module>
    from .session import Session
  File "/src/tools/c7n_azure/c7n_azure/session.py", line 27, in <module>
    from c7n_azure.utils import ResourceIdParser, StringUtils
  File "/src/tools/c7n_azure/c7n_azure/utils.py", line 20, in <module>
    from builtins import bytes
ImportError: No module named builtins```

Installing the future module fixes this. Can we put it out in the requirements.txt file?

Comment (kapilt) [reply Dockerfile]: thats odd thats a python3 only thing surprised that doesn't fail with py2 on ci

Comment (kapilt) [reply Dockerfile]: c7n_azure packages already has a dep on futures in its setup.py

**Comment by kapilt** [comment]
we should probably just move to py3 as a base, py2 has less than a year left of upstream support, poilcystream docker image does a nice job as well for py3 with multi-cloud custodian.

File: Dockerfile [hunk 2]
```diff
@@ -2,9 +2,21 @@ FROM python:2.7
...
 
+# Install Custodian Azure
+WORKDIR /src/tools/c7n_azure
+RUN pip install -r requirements.txt -e .
+RUN python setup.py develop
+
+# Install Custodian GCP
+WORKDIR /src/tools/c7n_gcp
+RUN pip install -r requirements.txt -e .
+RUN python setup.py develop
```
Comment (kapilt) [inline Dockerfile]: setup.py develop is redundant to install -e

**Comment by kapilt** [comment]
going to just close this in favor of a clean pr against the dockerfile #3336, using policystream as a base to switch to py3, add metadata, and multi-cloud.

**Comment by PratMis** [comment]
Cool, thanks!

**Pull request closed without merging**